src/webserver.cpp

```diff
@@ -402,6 +409,7 @@ MHD_Result policy_callback(void *cls, const struct sockaddr* addr, socklen_t add
 
     if (!(static_cast<webserver*>(cls))->ban_system_enabled) return MHD_YES;
 
+    std::shared_lock bans_and_allowances_lock((static_cast<webserver*>(cls))->bans_and_allowances_mutex);
```

Comment:
We can split the if statement here so that we always check only one of the two based on the chosen policy

Reply:
In both ACCEPT and REJECT cases, we access `bans` and `allowances`, so I don't see how we could only lock one of the mutexes.

I added a commit simplifying the code, with no intended functional change.

Reply:
[MINOR] I think the cleanest way would be to have two private methods (is_allowed and is_banned) that use the locks internally and just call into those methods from within here. That should localize the locks to the maximum extent possible.